pkg/authentication/x509/x509_test.go

```diff
@@ -0,0 +1,274 @@
+	t.Run("X509 agent", func(t *testing.T) {
+		auth := automationconfig.Auth{}
+		mdb := buildX509Configurable("mdb", mocks.BuildX509MongoDBUser("my-user"), mocks.BuildScramMongoDBUser("my-scram-user"))
+
+		agentSecret := CreateAgentCertificateSecret("tls.crt", mdb, false)
+		keyfileSecret := secret.Builder().
+			SetName(mdb.GetAgentKeyfileSecretNamespacedName().Name).
+			SetNamespace(mdb.GetAgentKeyfileSecretNamespacedName().Namespace).
+			SetField(constants.AgentKeyfileKey, "RuPeMaIe2g0SNTTa").
+			Build()
+		secrets := mocks.NewMockedSecretGetUpdateCreateDeleter(agentSecret, keyfileSecret)
+
+		err := Enable(&auth, secrets, mdb)
+		assert.NoError(t, err)
+
+		expected := automationconfig.Auth{
+			Users: []automationconfig.MongoDBUser{
+				{
+					Mechanisms: []string{},
+					Roles: []automationconfig.Role{
+						{
+							Role:     "readWrite",
+							Database: "admin",
+						},
+						{
+							Role:     "clusterAdmin",
+							Database: "admin",
+						},
+					},
+					Username:                   "CN=my-user,OU=organizationalunit,O=organization",
+					Database:                   "$external",
+					AuthenticationRestrictions: []string{},
+				},
+			},
+			Disabled:                 false,
+			AuthoritativeSet:         false,
+			AutoAuthMechanisms:       []string{constants.X509},
+			AutoAuthMechanism:        constants.X509,
+			DeploymentAuthMechanisms: []string{constants.X509},
+			AutoUser:                 "CN=mms-automation-agent,OU=ENG,O=MongoDB",
+			Key:                      "RuPeMaIe2g0SNTTa",
+			KeyFile:                  "/path/to/keyfile",
+			KeyFileWindows:           constants.AutomationAgentWindowsKeyFilePath,
+			AutoPwd:                  "",
+		}
+
+		assert.Equal(t, expected, auth)
+	})
+	t.Run("SCRAM agent", func(t *testing.T) {
+		auth := automationconfig.Auth{}
+		mdb := buildScramConfigurable("mdb", mocks.BuildX509MongoDBUser("my-user"), mocks.BuildScramMongoDBUser("my-scram-user"))
+
+		secrets := mocks.NewMockedSecretGetUpdateCreateDeleter()
+
+		err := Enable(&auth, secrets, mdb)
+		assert.NoError(t, err)
+
+		expected := automationconfig.Auth{
+			Users: []automationconfig.MongoDBUser{{
+				Mechanisms: []string{},
+				Roles: []automationconfig.Role{
+					{
+						Role:     "readWrite",
+						Database: "admin",
+					},
+					{
+						Role:     "clusterAdmin",
+						Database: "admin",
+					},
+				},
+				Username:                   "CN=my-user,OU=organizationalunit,O=organization",
+				Database:                   "$external",
+				AuthenticationRestrictions: []string{},
+			}},
+			Disabled:                 false,
+			AuthoritativeSet:         false,
+			DeploymentAuthMechanisms: []string{constants.X509},
+		}
+
+		assert.Equal(t, expected, auth)
+	})
+}
+
+func Test_ensureAgent(t *testing.T) {
+	auth := automationconfig.Auth{}
+	mdb := buildX509Configurable("mdb")
+	secrets := mocks.NewMockedSecretGetUpdateCreateDeleter()
+
+	err := ensureAgent(&auth, secrets, mdb)
+	assert.Error(t, err)
```

Comment:
do we care about the error type/return/msg?

Reply:
This specific error does not have a message, it has a reason "NotFound", however I can't seem to test for that. For the others, I will also check their message.

Reply:
Oh, I only meant to verify whether we care about the error msg. If we don't there is no need to test for it

Reply:
No, it makes sense to test for it. Good suggestion!